vmdb/app/models/ems_refresh/save_inventory_configuration.rb

```diff
@@ -0,0 +1,34 @@
+# Calling order for EmsForeman
+#
+# - ems
+#   - configuration_profiles (foreman HostGroup)
+#     - operating_system_flavor (links)
+#     - customization_script_ptable (link)
+#     - customization_script_medium (link)
+#   - configured_systems (foreman Host)
+#     - operating_system_flavor (links)
+#     - customization_scripts (links)
+## ? use guid?
+module EmsRefresh
+  module SaveInventoryConfiguration
+    def save_configuration_manager_inventory(manager, hashes, target = nil)
+      return if hashes.nil?
+      save_child_inventory(manager, hashes, [:configuration_profiles, :configured_systems], target)
+      manager.save
+    end
+
+    def save_configuration_profiles_inventory(manager, hashes, target)
+      hash_contains_all_values = target.nil? || manager == target
```

Comment:
That is the way I read the current logic: If the `target` is the `ems`, then we are loading all relations into the hash.

If the hash contains all children, then lets run through the `deleted = parent.assoc.dup` scenarios.

you like `delete_missing_records` better?